tests/app/main/views/test_suppliers.py

```diff
@@ -516,25 +516,26 @@ def test_should_be_a_503_if_email_fails(self, data_api_client, DMNotifyClient):
 
 @mock.patch('app.main.views.suppliers.data_api_client')
 class TestUpdatintSupplierName(LoggedInApplicationTest):
-    def test_admin_can_test_update_supplier_name(self, data_api_client):
-        data_api_client.get_supplier.return_value = {"suppliers": {"id": 1234, "name": "Something Old"}}
-        response = self.client.post(
-            '/admin/suppliers/1234/edit/name',
-            data={'new_supplier_name': 'Something New'}
-        )
-        assert response.status_code == 302
-        assert response.location == 'http://localhost/admin/suppliers?supplier_id=1234'
-        data_api_client.update_supplier.assert_called_once_with(1234, {'name': "Something New"}, "test@example.com")
-
-    def test_ccs_roles_can_not_update_supplier_name(self, data_api_client):
-        for role in ('admin-ccs-sourcing', 'admin-ccs-category'):
+    def test_admin_and_ccs_category_roles_can_update_supplier_name(self, data_api_client):
+        for role in ('admin', 'admin-ccs-category'):
             self.user_role = role
+            data_api_client.get_supplier.return_value = {"suppliers": {"id": 1234, "name": "Something Old"}}
             response = self.client.post(
                 '/admin/suppliers/1234/edit/name',
                 data={'new_supplier_name': 'Something New'}
             )
-            assert response.status_code == 403
-            assert data_api_client.update_supplier.called is False
+            assert response.status_code == 302
+            assert response.location == 'http://localhost/admin/suppliers?supplier_id=1234'
+            data_api_client.update_supplier.assert_called_with(1234, {'name': "Something New"}, "test@example.com")
+
+    def test_ccs_sourcing_role_can_not_update_supplier_name(self, data_api_client):
+        self.user_role = 'admin-ccs-sourcing'
+        response = self.client.post(
+            '/admin/suppliers/1234/edit/name',
+            data={'new_supplier_name': 'Something New'}
+        )
+        assert response.status_code == 403
+        assert data_api_client.update_supplier.called is False
```

Comment:
I'd prefer `assert_not_called` (well, really I'd prefer `assert foo.call_args_list == []` everywhere but no-one seems keen on that style apart from me!) however we do use this style elsewhere in this module. So up to you if you want to change.

Reply:
Rob also likes the `call_args_list`.  I'm ambivalent.

This is only green in the diff because of an indentation change where I removed the loop... not really new code.

Reply:
But I can fix it while I'm here now.